pilot/pkg/bootstrap/server.go

```diff
@@ -573,9 +518,13 @@ func (s *Server) initSecureGrpcServerDNS(addr string) error {
 			// on a listener
 			err := s.secureHTTPServerDNS.ServeTLS(secureGrpcListener, "", "")
 			msg := fmt.Sprintf("Stoppped listening on %s %v", dnsGrpc, err)
-			<-stop
-			log.Info(msg)
-			if s.Args.ForceStop {
+			select {
+			case <-stop:
+				log.Info(msg)
+			default:
+				panic(fmt.Sprintf("%s due to error: %v", msg, err))
```

Comment:
won't this panic if we haven't sent to stop yet? I thought the point was we block until send to stop?

Reply:
No, s.secureHTTPServerDNS.ServeTLS() will only return when the server is closed or an error occur. 
But thanks for the info, there is a bug originally, the L527-L535 should be in a separate routine waiting on stop channel

Reply:
fixed

Reply:
Then what is the point of panic? Why do we even need this?

Reply:
This is to handle  `ServeTLS` error return when bootstrap, we should have a way to notify pilot exit. BTW, this is legacy logic